server/verify.go

```diff
@@ -138,33 +136,40 @@ func pubKeysEqual(k1 crypto.PublicKey, k2 crypto.PublicKey) bool {
 }
 
 // Checks if the provided AK public key can be trusted
-func checkAkTrusted(ak crypto.PublicKey, opts VerifyOpts) error {
-	if len(opts.TrustedAKs) == 0 {
-		return fmt.Errorf("no mechanism for AK verification provided")
+func checkAKTrusted(ak crypto.PublicKey, akCertBytes []byte, opts VerifyOpts) error {
+	checkPub := len(opts.TrustedAKs) > 0
+	checkCert := len(opts.TrustedRootCerts.Subjects()) > 0
```

Comment:
the tests are failing (https://github.com/google/go-tpm-tools/runs/4812554993?check_suite_focus=true) because you need to check that opts.TrustedRootCerts != nil

Reply:
Fixed